src/vm/tests/simple_apply_eval.rs

```diff
@@ -541,6 +541,196 @@ fn test_simple_arithmetic_functions() {
+    // Tests the sequence comparisons against ClarityVersion2. The new kinds of
+    // sequence comparison *should* work.
+    let success_tests = [
+        (r#"(is-eq "aaa" "aaa")"#, Value::Bool(true)),
+        (r#"(is-eq "aba" "aaa")"#, Value::Bool(false)),
+        (r#"(is-eq u"aaa" u"aaa")"#, Value::Bool(true)),
+        (r#"(is-eq u"aba" u"aaa")"#, Value::Bool(false)),
+        (r#"(is-eq 0x010203 0x010203)"#, Value::Bool(true)),
+        (r#"(is-eq 0x090203 0x010203)"#, Value::Bool(false)),
+        (r#"(< 0x0100 0x0100)"#, Value::Bool(false)),
+        (r#"(< 0x0101 0x010101)"#, Value::Bool(true)),
+        (r#"(< 0x010101 0x0101)"#, Value::Bool(false)),
+        (r#"(< "aaa" "aaa")"#, Value::Bool(false)),
+        (r#"(< "aa" "aaa")"#, Value::Bool(true)),
+        (r#"(< "aaa" "aa")"#, Value::Bool(false)),
+        (r#"(< u"aaa" u"aaa")"#, Value::Bool(false)),
+        (r#"(< u"aa" u"aaa")"#, Value::Bool(true)),
+        (r#"(< u"aaa" u"aa")"#, Value::Bool(false)),
+        (r#"(<= 0x0100 0x0100)"#, Value::Bool(true)),
+        (r#"(<= 0x0101 0x010101)"#, Value::Bool(true)),
+        (r#"(<= 0x010101 0x0101)"#, Value::Bool(false)),
+        (r#"(<= "aaa" "aaa")"#, Value::Bool(true)),
+        (r#"(<= "aa" "aaa")"#, Value::Bool(true)),
+        (r#"(<= "aaa" "aa")"#, Value::Bool(false)),
+        (r#"(<= u"aaa" u"aaa")"#, Value::Bool(true)),
+        (r#"(<= u"aa" u"aaa")"#, Value::Bool(true)),
+        (r#"(<= u"aaa" u"aa")"#, Value::Bool(false)),
+        (r#"(> 0x0100 0x0100)"#, Value::Bool(false)),
+        (r#"(> 0x0101 0x010101)"#, Value::Bool(false)),
+        (r#"(> 0x010101 0x0101)"#, Value::Bool(true)),
+        (r#"(> "aaa" "aaa")"#, Value::Bool(false)),
+        (r#"(> "aa" "aaa")"#, Value::Bool(false)),
+        (r#"(> "aaa" "aa")"#, Value::Bool(true)),
+        (r#"(> u"aaa" u"aaa")"#, Value::Bool(false)),
+        (r#"(> u"aa" u"aaa")"#, Value::Bool(false)),
+        (r#"(> u"aaa" u"aa")"#, Value::Bool(true)),
+        (r#"(>= 0x0100 0x0100)"#, Value::Bool(true)),
+        (r#"(>= 0x0101 0x010101)"#, Value::Bool(false)),
+        (r#"(>= 0x010101 0x0101)"#, Value::Bool(true)),
+        (r#"(>= "aaa" "aaa")"#, Value::Bool(true)),
+        (r#"(>= "aa" "aaa")"#, Value::Bool(false)),
+        (r#"(>= "aaa" "aa")"#, Value::Bool(true)),
+        (r#"(>= u"aaa" u"aaa")"#, Value::Bool(true)),
+        (r#"(>= u"aa" u"aaa")"#, Value::Bool(false)),
+        (r#"(>= u"aaa" u"aa")"#, Value::Bool(true)),
+    ];
+
+    // Note: Execute against Clarity2.
+    success_tests.iter().for_each(|(program, expectation)| {
+        assert_eq!(
+            expectation.clone(),
+            execute_v2(program),
+            "{:?}, {:?}",
+            program,
+            expectation.clone()
+        )
+    });
+}
+
+#[test]
+fn test_sequence_comparisons_mismatched_types() {
+    // Tests that comparing objects of different types results in an error in Clarity1.
+    let error_tests = ["(> 0 u1)", "(< 0 u1)"];
```

Comment:
should evaluate these tests in clarity2 as well

Reply:
ok.. good idea.